docs/user/guides/setup/configuration-file.rst

```diff
@@ -0,0 +1,223 @@
+or in the future when you need features from a newer Ubuntu.
+
+.. seealso::
+
+   :ref:`config-file/v2:build.os`
+     Configuration file reference with all values possible for ``build.os``.
+
+
+Adjust: Python configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using Python in your builds,
+you should define the Python version in ``build.tools.python``.
+
+The ``python`` key contains a list of sub-keys,
+specifying the requirements to install.
+
+- Use ``python.install.package`` to install the project itself as a Python package using pip
+- Use ``python.install.requirements`` to install packages from a requirements file
+- Use ``build.jobs`` to install packages using Poetry or PDM
+
+.. seealso::
+
+   :ref:`config-file/v2:build.tools.python`
+     Configuration file reference with all Python versions available for ``build.tools.python``.
+
+   :ref:`config-file/v2:python`
+     Configuration file reference for configuring the Python environment activated by ``build.tools.python``.
+
+Adjust: Sphinx and MkDocs version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using either the ``sphinx`` or ``mkdocs`` builder,
+then Sphinx or MkDocs will be installed automatically in its latest version.
+
+But we recommend that you specify the version that your documentation project uses.
+The ``requirements`` key is a file path that points to a text (``.txt``) file
+that lists the Python packages you want Read the Docs to install.
+
+.. seealso::
+
+   :ref:`guides/reproducible-builds:Use a requirements file for Python dependencies`
+      This guide explains how to specify Python requirements,
+      such as the version of Sphinx or MkDocs.
+
+   :ref:`config-file/v2:sphinx`
+     Configuration file reference for configuring the Sphinx builder.
+
+   :ref:`config-file/v2:mkdocs`
+     Configuration file reference for configuring the MkDocs builder.
+
+.. _config_howto_build.tools:
+
+Adjust: Additional build environments?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you need software from additional platforms (nodejs, rust, golang),
+you can add and enable such an environment so you can install the packages you need.
+
+.. seealso::
+
+   :ref:`config-file/v2:build.tools`
+     List of all tools that are possible to enable.
+
+Adjust: Custom tools
+~~~~~~~~~~~~~~~~~~~~
+
+It's possible to build documentation using almost any documentation tool,
+as long as an environment is available (see :ref:`the previous step <config_howto_build.tools>`).
+Adding your own build commands is done by listing them in ``build.commands``.
+
+The :ref:`Other tools <howto_templates>` template shows an example of how to do that.
+
+.. seealso::
+
+   :doc:`/build-customization`
+     Jump to a full guide explaining how to setup (almost) any documentation tool.
+
+   :ref:`config-file/v2:build.commands`
+     Configuration file reference for custom build commands.
+
```

Comment:
I probably have the same feeling here, `build.commands` is not a good next step for novice users. The progression should try to balance giving the user control and us retaining some control, so: supported tool (sphinx/mkdocs) -> build jobs customization (intermediate, configuration file still works) -> build commands (expert, you're on your own).

Perhaps instead of jumping right to "custom tools" we should be pointing towards "custom jobs" and customizing the build process with new jobs. This at least serves us to promote advanced usage and make the user aware of things without through them into the gauntlet of build.commands.

Reply:
I killed this section now!

I think the method that works is to have it all in a tab "Other tools" further up. So all this information should only be seen by users that aren't using Sphinx, MkDocs nor Docusaurus. It might be more _advanced_, but I think that it's better than leading them to a dead-end with no information available.

Reply:
That's fair for now too. We do link out to the build customization page already, so perhaps that is what I am describing as well.